transformers/pipelines.py

```diff
@@ -94,32 +95,71 @@ def __call__(self, *texts, **kwargs):
 
 
 class QuestionAnsweringPipeline(Pipeline):
+    """
+    Question Answering pipeling involving Tokenization and Inference.
+    TODO:
+     - top-k answers
+     - return start/end chars
+     - return score
+    """
+
+    def __init__(self, model, tokenizer: Optional[PreTrainedTokenizer]):
+        super().__init__(model, tokenizer)
+
+    @staticmethod
+    def create_sample(question: Union[str, List[str]], context: Union[str, List[str]]) -> Union[dict, List[Dict]]:
+        is_list = isinstance(question, list)
+
+        if is_list:
+            return [{'question': q, 'context': c} for q, c in zip(question, context)]
+        else:
+            return {'question': question, 'context': context}
 
     @classmethod
     def from_config(cls, model, tokenizer: PreTrainedTokenizer, **kwargs):
         pass
 
-    def __call__(self, texts, **kwargs):
+    def __call__(self, *texts, **kwargs):
+        # Set defaults values
+        kwargs.setdefault('max_answer_len', 15)
+        kwargs.setdefault('topk', 1)
+
+        if kwargs['topk'] < 1:
+            raise ValueError('topk parameter should be >= 1 (got {})'.format(kwargs['topk']))
+
+        if kwargs['max_answer_len'] < 1:
+            raise ValueError('max_answer_len parameter should be >= 1 (got {})'.format(kwargs['max_answer_len']))
+
+        # Tabular input
+        if 'question' in kwargs and 'context' in kwargs:
+            texts = QuestionAnsweringPipeline.create_sample(kwargs['question'], kwargs['context'])
+        elif 'data' in kwargs:
+            texts = kwargs['data']
         # Generic compatibility with sklearn and Keras
-        if 'X' in kwargs and not texts:
+        elif 'X' in kwargs and not texts:
             texts = kwargs.pop('X')
+        else:
+            (texts, ) = texts
 
-        if not isinstance(texts, (tuple, list)):
-            raise Exception('QuestionAnsweringPipeline requires predict argument to be a tuple (context, question) or a List of tuple.')
+        if not isinstance(texts, (dict, list)):
+            raise Exception('QuestionAnsweringPipeline requires predict argument to be a tuple (context, question) or a List of dict.')
 
         if not isinstance(texts, list):
             texts = [texts]
 
+        # Map to tuple (question, context)
+        texts = [(text['question'], text['context']) for text in texts]

```

Comment:
here we will actually need to fallback on @LysandreJik SQuAD processing methods added in #1984 (should be finished) because some models have question before context and other model do it the other way around 🙃

Reply:
Make sense, I based the actual ordering w.r.t the current fine-tuned models we had. I'll move to Lysandre processing.

Reply:
Done in 40a39ab

Reply:
And 9a24e0c